src/content/drafts/cesium-eslint-case-study.md

```diff
@@ -0,0 +1,41 @@
+authors:
+  - sanjeetsuhag
+categories:
+  - Case Studies
+tags:
+  - Guest Post
+  - Rules
+---
+
+## Background
+
+[CesiumJS](https://cesium.com/platform/cesiumjs/) is an open source JavaScript library for 3D geospatial visualization. Cesium’s roots are in aerospace with our first use case being accurate visualization of satellites and spacecraft. Now with more than 4,000,000 downloads and 9000 stars on GitHub, CesiumJS has a large community of users and contributors. Users from all over the world have used CesiumJS for a variety of different applications, from mapping and data analysis to [building interactive simulations for satellites](https://cesium.com/blog/2022/08/11/sedaro-satellite-visualizes-spacecraft-digital-twins-cesiumjs/), [event planning](https://cesium.com/blog/2022/07/27/oneplan-and-cesium-delivering-geospatial-for-events/), [open source intelligence](https://cesium.com/blog/2022/03/08/cesium-story-depicts-damage-in-ukraine/) and even for building [flight simulators](https://cesium.com/blog/2021/12/06/geofs-is-a-flight-simulator-that-showcases-global-satellite/).
+
+## Switching to ESLint
+
+As an open source project with frequent external contributions, it is important that we have clear and consistent coding conventions along with automated ways of enforcing them. Development started in 2011, and the first version of Cesium was released in 2014. As new browser APIs and standards have emerged, ensuring that CesiumJS stays compatible with modern use cases has been very important to us.
+
+Initially, CesiumJS utilized JSHint for static analysis, since it was the standard at the time. In 2017, we switched over to ESLint since it was better at keeping pace with the rapid nature of ECMAScript standards development. It enabled enforcement of style rules through linting and applying automatic fixes for a lot of those rules. Perhaps most important of all was the ability to create extensible configurations that fit our own coding standards and extend configurations made by others in the community. In this post, I will explain how we, the maintainers of CesiumJS, use ESLint for ensuring code quality, enforcing consistency, and reducing developer burden in adhering to our code style and standards across our open source and private projects.
+
+
+## How we use ESLint
+
+The earlier in the development process a developer is able to catch and fix an issue, the less expensive it is to fix it. This “shifting left” strategy is a key priority for us and that is why we incorporate linting into the developer workflow so that developers are informed of common potential issues in their code as they write it or before they commit it. First, Visual Studio Code is the IDE of choice at Cesium, so we include the official ESLint VS Code extension in .vscode/extensions.json so that users are prompted to install the extension when they clone the repository and open it in VS Code. 
```

Comment:
```suggestion
The earlier in the development process a developer is able to catch an issue, the less expensive it is to fix it. This “shifting left” strategy is a key priority for us and that is why we incorporate linting into the developer workflow so that developers are informed of potential issues in their code as they write it or before they commit it. 

First, Visual Studio Code is the IDE of choice at Cesium, so we include the official ESLint VS Code extension in `.vscode/extensions.json` so that users are prompted to install the extension when they clone the repository and open it in VS Code. 
```